src/Engine/AssetsManager.cpp

```diff
@@ -1,15 +1,41 @@
 #include "Engine/AssetsManager.h"
 
 #include <algorithm>
+#include <filesystem>
 
 #include "Engine/LOD.h"
 
 #include "Engine/Graphics/IRender.h"
 #include "Engine/Graphics/ImageLoader.h"
 #include "Engine/Graphics/Texture.h"
 #include "GUI/GUIFont.h"
+#include "Library/Lod/LodReader.h"
+
+
+std::unique_ptr<AssetsManager> assets = nullptr;
+
+
+std::unique_ptr<AssetsManager> AssetsManager::create(const std::string &resources_path) {
```

Comment:
Let's just use a plain constructor instead of a factory method.

Factory methods make sense when what you're returning might be a member of a diverse hierarchy, and the actual type is not known at call site.

This is not the case here.

Reply:
OK I've just realized you're doing this because it provides an error-handling mechanism different from exception handling.